_posts/2019-07-15-writing-fast-rails-part-3.markdown

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title:  "Tips for Writing Fast Rails: Part 3"
+date: 2019-07-15 08:00
+categories: ["rails", "performance"]
+author: "luciano"
+---
+
+Here we continue with the [series](https://fastruby.io/blog/tags/performance) of articles where we talk how minor adjustments in the code can lead to major performance improvements.
+
+In this article we'll focus on the use of [ActiveRecord::Batches#find_each](https://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each) when it comes to iterations across a large number of records.
+
+<!--more-->
+
+## Prefer `Model.find_each` instead of `Model.all.each`
+
+A very common mistake (often seen in background jobs) is to use [ActiveRecord::Scoping::Named::ClassMethods#all](https://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html#method-i-all) + [ActiveRecord::Result#each](https://api.rubyonrails.org/classes/ActiveRecord/Result.html#method-i-each) (`Model.all.each`) to iterate a table with several thousands of records on it.
```

Comment:
@lubc Maybe somewhere in this article you could mention this project? https://github.com/rubocop-hq/rubocop-performance/blob/c0ce42bc72f256c9a07585569532f4f754adfbd0/lib/rubocop/cop/performance/count.rb

I believe this could be an interesting performance cop for that rubocop performance project.

Reply:
@etagwerker Sure 👍 just added a link to that at the end